robotframework-ls/src/robotframework_ls/impl/semantic_tokens.py

```diff
@@ -92,14 +93,53 @@ def extract_gherkin_token_from_keyword(keyword_token):
     return gherkin_token
 
 
-def extract_library_token_from_keyword(keyword_token):
+def extract_library_token_from_keyword(keyword_token, context):
+    if not "." in keyword_token.value:
+        return None
     library_token = None
-    library_token_length = keyword_token.value.rfind(".")
-    if library_token_length > 0:
-        library_token = keyword_token.extract_token("name", library_token_length)
+    library_references = get_list_of_potential_library_names_from_keyword(
+        keyword_token.value
+    )
+    imported_libraries = get_list_of_library_names_from_settings(context)
+    for reference in library_references:
+        if reference in imported_libraries:
+            library_token = keyword_token.extract_token("name", len(reference))
+            break
     return library_token
 
 
+def get_list_of_potential_library_names_from_keyword(keyword_name):
+    potential_library_names = []
+    name_length = -1
+    while True:
+        name_length = keyword_name.find(".", name_length + 1)
+        if name_length == -1:
+            break
+        library_name = keyword_name[:name_length].lower()
+        potential_library_names.append(library_name)
+    return potential_library_names
+
+
+def get_list_of_library_names_from_settings(context):
+    from robot.api import Token
+    import os
+
+    library_names = []
+    append = library_names.append
+
+    for library_import in context.get_imported_libraries():
+        for token in library_import:
+            if token.type == Token.NAME:
```

Comment:
I'm failing to see how WITH_NAME is handled here... does it fall under `Token.NAME` too?

Reply:
Yes it does